src/anyio/_backends/_trio.py

```diff
@@ -500,20 +500,27 @@ async def release(self) -> None:
         self._lock.release()
 
 
-class Event:
+class BaseEvent(abc.BaseEvent):
     def __init__(self):
         self._event = trio.Event()
 
-    async def set(self) -> None:
-        self._event.set()
-
     def is_set(self) -> bool:
         return self._event.is_set()
 
     async def wait(self):
         await self._event.wait()
 
 
+class Event(abc.Event, BaseEvent):
+    async def set(self) -> None:
+        self._event.set()
```

Comment:
A rule of trio api's is that async function should checpoint on all code paths, so add sleep(0). Original code had the same problem

Reply:
Good point. :+1: I guess that an `trio.lowlevel.checkpoint()` is even more descriptive than a raw `sleep(0)`.